File: remove-element-1.scala
object Solution {
    def removeElement(nums: Array[Int], `val`: Int): Int = {
        var size = 0
        
        for (i <- 0 until nums.length) {
            if (nums(i) != `val`) {
                nums(size) = nums(i)
                size += 1
            }
        }
        
        size
    }
}

File: remove-element-1.java
class Solution {
    public int removeElement(int[] nums, int val) {
        
        int p1 = 0, p2 = 0;
        
        for ( ; p1 < nums.length; p1++) {
            if (nums[p1] == val)
                continue;
            
            nums[p2] = nums[p1];
            p2++;
        }
        
        return p2;
    }
}
